### PR TITLE
Add clippy to Travis and remove unnecessary code

### PR DESCRIPTION
**Pull request by romac**
On hold. See https://github.com/SpinResearch/RustySecrets/pull/66#issuecomment-391170134.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -21,6 +21,15 @@ matrix:
...
         - cargo fmt --all -- --write-mode=diff
+    - env:
+        - NAME='clippy'
+        # use env so updating versions causes cache invalidation
+        - CLIPPY_VERSION=0.0.200
+      rust: nightly-2018-05-14
+      before_script:
+        - cargo install clippy --version $CLIPPY_VERSION || echo "clippy already installed"
+      script:
+        - cargo clippy --all -- -D clippy
```
Comment (psivesely) [inline .travis.yml]: `- cargo clippy --all --tests --all-features -- -D clippy`

**Comment by nvesely** [comment]
Maybe wait on this until a unified Travis testing strategy is figured out for all the RustySecrets related repositories (see https://github.com/SpinResearch/merkle.rs/pull/36#discussion_r190078423).

**Comment by romac** [comment]
@nvesely Agreed.

**Comment by afck** [comment]
Clippy can now also be installed via rustup (instead of `cargo install`), so you don't have to manually match Clippy versions to the corresponding Rust nightlies anymore. And it will even be part of one of the upcoming stable versions! :tada: 
https://internals.rust-lang.org/t/clippy-is-available-as-a-rustup-component/7967

**Comment by psivesely** [comment]
That simplifies things a lot. Ideally our CI is simply not as complicated as some of the ideas we were discussing a month or two back :smiley_cat:

**Pull request closed without merging**